src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -0,0 +1,112 @@
+package bio.terra.service.search;
+
+import bio.terra.common.exception.PdaoException;
+import bio.terra.model.SearchIndexModel;
+import bio.terra.model.SearchIndexRequest;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.CreateIndexResponse;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.elasticsearch.client.indices.GetIndexResponse;
+import org.elasticsearch.client.indices.PutMappingRequest;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class SearchService {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final RestHighLevelClient client;
+
+    @Autowired
+    public SearchService(BigQueryPdao bigQueryPdao) {
+        // needs to be moved into own class with config
+        final RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200));
+        this.bigQueryPdao = bigQueryPdao;
+        this.client = new RestHighLevelClient(builder);
+    }
+
+    private String createEmptyIndex(Snapshot snapshot) {
+        String indexName = String.format("idx-%s", snapshot.getId());
+        try {
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest(indexName)
+                .settings(Settings.builder()
+                    .put("index.number_of_shards", 3));
+            CreateIndexResponse response = client.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+            if (!response.isAcknowledged()) {
+                throw new PdaoException("The index request was not acknowledged by one or more nodes");
+            }
+        } catch (IOException e) {
+            throw new PdaoException("Error creating index", e);
+        }
+        return indexName;
+    }
+
+    private void createIndexMapping(String indexName, List<Map<String, Object>> values) {
+        PutMappingRequest request = new PutMappingRequest(indexName);
+        Map<String, Object> properties = new HashMap<>();
+        values.get(0).forEach((key, val) -> {
+            properties.put(key, Map.of("type", "text"));
+        });
```

Comment:
Not sure if it's any better, but this can be done using streaming. I think this is correct:
```suggestion
        Map<String, Object> properties =
            values.get(0).keySet().stream()
            .collect(Collectors.toMap(Function.identity(), x -> Map.of("type", "text")));
```
The only thing I like better about the streaming version is it's clearer (?) that this is a `map` operation over a `Map`. Before it was less clear (to me) that, for example, you're making a new map that's the same size as the old one.

Reply:
I switched to using this approach.